### PR TITLE
[eclipse/xtext#1384] Convert to Declarative Pipeline

### PR DESCRIPTION
**Pull request by kthoms**
- add Slack notification
- externalize build call to shell script

Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

**Comment by kthoms** [comment]
Please wait for build

File: Jenkinsfile [hunk 1]
```diff
@@ -1,73 +1,107 @@
...
+          if [ -d ".git" ]; then
+            git reset --hard
+          fi
+        '''
+        checkout scm
+        
+        script {
+          if ("latest" == params.TARGET_PLATFORM) {
+            currentBuild.displayName = "#${BUILD_NUMBER}(x)"
+          } else if ("r201903" == params.target_platform) {
```
Comment (szarnekow) [inline Jenkinsfile]: I don't see r201903 in the list of choices above

**Comment by cdietrich** [comment]
@kthoms @szarnekow anything left?

**Comment by cdietrich** [comment]
ping @kthoms

**Pull request merged**